2025-08-27 17:24:44,023 INFO    MainThread:1563477 [wandb_setup.py:_flush():80] Current SDK version is 0.21.1
2025-08-27 17:24:44,023 INFO    MainThread:1563477 [wandb_setup.py:_flush():80] Configure stats pid to 1563477
2025-08-27 17:24:44,024 INFO    MainThread:1563477 [wandb_setup.py:_flush():80] Loading settings from /home/ubuntu/.config/wandb/settings
2025-08-27 17:24:44,024 INFO    MainThread:1563477 [wandb_setup.py:_flush():80] Loading settings from /home/ubuntu/gpt_oss_arc_final/wandb/settings
2025-08-27 17:24:44,024 INFO    MainThread:1563477 [wandb_setup.py:_flush():80] Loading settings from environment variables
2025-08-27 17:24:44,024 INFO    MainThread:1563477 [wandb_init.py:setup_run_log_directory():703] Logging user logs to /home/ubuntu/gpt_oss_arc_final/wandb/run-20250827_172444-17nljko7/logs/debug.log
2025-08-27 17:24:44,024 INFO    MainThread:1563477 [wandb_init.py:setup_run_log_directory():704] Logging internal logs to /home/ubuntu/gpt_oss_arc_final/wandb/run-20250827_172444-17nljko7/logs/debug-internal.log
2025-08-27 17:24:44,024 INFO    MainThread:1563477 [wandb_init.py:init():830] calling init triggers
2025-08-27 17:24:44,024 INFO    MainThread:1563477 [wandb_init.py:init():835] wandb.init called with sweep_config: {}
config: {'model': 'openai/gpt-oss-20b', 'learning_rate': 1e-05, 'batch_size': 1, 'num_problems': 3, 'reward_components': 4, '_wandb': {}}
2025-08-27 17:24:44,024 INFO    MainThread:1563477 [wandb_init.py:init():871] starting backend
2025-08-27 17:24:44,247 INFO    MainThread:1563477 [wandb_init.py:init():874] sending inform_init request
2025-08-27 17:24:44,254 INFO    MainThread:1563477 [wandb_init.py:init():882] backend started and connected
2025-08-27 17:24:44,255 INFO    MainThread:1563477 [wandb_init.py:init():953] updated telemetry
2025-08-27 17:24:44,256 INFO    MainThread:1563477 [wandb_init.py:init():977] communicating run to backend with 90.0 second timeout
2025-08-27 17:24:47,236 INFO    MainThread:1563477 [wandb_init.py:init():1029] starting run threads in backend
2025-08-27 17:24:47,429 INFO    MainThread:1563477 [wandb_run.py:_console_start():2494] atexit reg
2025-08-27 17:24:47,430 INFO    MainThread:1563477 [wandb_run.py:_redirect():2342] redirect: wrap_raw
2025-08-27 17:24:47,430 INFO    MainThread:1563477 [wandb_run.py:_redirect():2411] Wrapping output streams.
2025-08-27 17:24:47,430 INFO    MainThread:1563477 [wandb_run.py:_redirect():2434] Redirects installed.
2025-08-27 17:24:47,435 INFO    MainThread:1563477 [wandb_init.py:init():1075] run started, returning control to user process
2025-08-27 17:30:15,847 INFO    MainThread:1563477 [wandb_run.py:_config_callback():1380] config_cb None None {'peft_config': {'default': {'task_type': <TaskType.CAUSAL_LM: 'CAUSAL_LM'>, 'peft_type': <PeftType.LORA: 'LORA'>, 'auto_mapping': None, 'base_model_name_or_path': 'openai/gpt-oss-20b', 'revision': None, 'inference_mode': False, 'r': 16, 'target_modules': {'q_proj', 'k_proj', 'up_proj', 'down_proj', 'o_proj', 'v_proj', 'gate_proj'}, 'exclude_modules': None, 'lora_alpha': 32, 'lora_dropout': 0.1, 'fan_in_fan_out': False, 'bias': 'none', 'use_rslora': False, 'modules_to_save': None, 'init_lora_weights': True, 'layers_to_transform': None, 'layers_pattern': None, 'rank_pattern': {}, 'alpha_pattern': {}, 'megatron_config': None, 'megatron_core': 'megatron.core', 'trainable_token_indices': None, 'loftq_config': {}, 'eva_config': None, 'corda_config': None, 'use_dora': False, 'use_qalora': False, 'qalora_group_size': 16, 'layer_replication': None, 'runtime_config': {'ephemeral_gpu_offload': False}, 'lora_bias': False, 'target_parameters': None}}, 'vocab_size': 201088, 'hidden_size': 2880, 'intermediate_size': 2880, 'num_hidden_layers': 24, 'num_attention_heads': 64, 'num_local_experts': 32, 'sliding_window': 128, 'num_experts_per_tok': 4, 'num_key_value_heads': 8, 'hidden_act': 'silu', 'initializer_range': 0.02, 'rms_norm_eps': 1e-05, 'rope_theta': 150000, 'rope_scaling': {'beta_fast': 32.0, 'beta_slow': 1.0, 'factor': 32.0, 'original_max_position_embeddings': 4096, 'rope_type': 'yarn', 'truncate': False}, 'attention_dropout': 0.0, 'head_dim': 64, 'layer_types': ['sliding_attention', 'full_attention', 'sliding_attention', 'full_attention', 'sliding_attention', 'full_attention', 'sliding_attention', 'full_attention', 'sliding_attention', 'full_attention', 'sliding_attention', 'full_attention', 'sliding_attention', 'full_attention', 'sliding_attention', 'full_attention', 'sliding_attention', 'full_attention', 'sliding_attention', 'full_attention', 'sliding_attention', 'full_attention', 'sliding_attention', 'full_attention'], 'attention_bias': True, 'max_position_embeddings': 131072, 'router_aux_loss_coef': 0.9, 'output_router_logits': False, 'use_cache': True, 'return_dict': True, 'output_hidden_states': False, 'torchscript': False, 'torch_dtype': 'bfloat16', 'pruned_heads': {}, 'tie_word_embeddings': False, 'chunk_size_feed_forward': 0, 'is_encoder_decoder': False, 'is_decoder': False, 'cross_attention_hidden_size': None, 'add_cross_attention': False, 'tie_encoder_decoder': False, 'architectures': ['GptOssForCausalLM'], 'finetuning_task': None, 'id2label': {0: 'LABEL_0', 1: 'LABEL_1'}, 'label2id': {'LABEL_0': 0, 'LABEL_1': 1}, 'task_specific_params': None, 'problem_type': None, 'tokenizer_class': None, 'prefix': None, 'bos_token_id': None, 'pad_token_id': 199999, 'eos_token_id': 200002, 'sep_token_id': None, 'decoder_start_token_id': None, 'max_length': 20, 'min_length': 0, 'do_sample': False, 'early_stopping': False, 'num_beams': 1, 'num_beam_groups': 1, 'diversity_penalty': 0.0, 'temperature': 1.0, 'top_k': None, 'top_p': 1.0, 'typical_p': 1.0, 'repetition_penalty': 1.0, 'length_penalty': 1.0, 'no_repeat_ngram_size': 0, 'encoder_no_repeat_ngram_size': 0, 'bad_words_ids': None, 'num_return_sequences': 1, 'output_scores': False, 'return_dict_in_generate': False, 'forced_bos_token_id': None, 'forced_eos_token_id': None, 'remove_invalid_values': False, 'exponential_decay_length_penalty': None, 'suppress_tokens': None, 'begin_suppress_tokens': None, '_name_or_path': 'openai/gpt-oss-20b', 'transformers_version': '4.55.3', 'experts_per_token': 4, 'initial_context_length': 4096, 'model_type': 'gpt_oss', 'swiglu_limit': 7.0, 'tf_legacy_loss': False, 'use_bfloat16': False, 'output_attentions': False, 'output_dir': '/data/gpt_oss_final/checkpoints_hf_trl_dapo', 'overwrite_output_dir': False, 'do_train': False, 'do_eval': False, 'do_predict': False, 'eval_strategy': 'no', 'prediction_loss_only': False, 'per_device_train_batch_size': 1, 'per_device_eval_batch_size': 8, 'per_gpu_train_batch_size': None, 'per_gpu_eval_batch_size': None, 'gradient_accumulation_steps': 4, 'eval_accumulation_steps': None, 'eval_delay': 0, 'torch_empty_cache_steps': None, 'learning_rate': 1e-05, 'weight_decay': 0.0, 'adam_beta1': 0.9, 'adam_beta2': 0.999, 'adam_epsilon': 1e-08, 'max_grad_norm': 1.0, 'num_train_epochs': 1, 'max_steps': 100, 'lr_scheduler_type': 'linear', 'lr_scheduler_kwargs': {}, 'warmup_ratio': 0.0, 'warmup_steps': 5, 'log_level': 'passive', 'log_level_replica': 'warning', 'log_on_each_node': True, 'logging_dir': '/data/gpt_oss_final/checkpoints_hf_trl_dapo/runs/Aug27_17-26-51_gpu-1', 'logging_strategy': 'steps', 'logging_first_step': False, 'logging_steps': 1, 'logging_nan_inf_filter': True, 'save_strategy': 'steps', 'save_steps': 10, 'save_total_limit': None, 'save_safetensors': True, 'save_on_each_node': False, 'save_only_model': False, 'restore_callback_states_from_checkpoint': False, 'no_cuda': False, 'use_cpu': False, 'use_mps_device': False, 'seed': 42, 'data_seed': None, 'jit_mode_eval': False, 'use_ipex': False, 'bf16': True, 'fp16': False, 'fp16_opt_level': 'O1', 'half_precision_backend': 'auto', 'bf16_full_eval': False, 'fp16_full_eval': False, 'tf32': None, 'local_rank': 0, 'ddp_backend': None, 'tpu_num_cores': None, 'tpu_metrics_debug': False, 'debug': [], 'dataloader_drop_last': False, 'eval_steps': None, 'dataloader_num_workers': 0, 'dataloader_prefetch_factor': None, 'past_index': -1, 'run_name': None, 'disable_tqdm': False, 'remove_unused_columns': False, 'label_names': None, 'load_best_model_at_end': False, 'metric_for_best_model': None, 'greater_is_better': None, 'ignore_data_skip': False, 'fsdp': [], 'fsdp_min_num_params': 0, 'fsdp_config': {'min_num_params': 0, 'xla': False, 'xla_fsdp_v2': False, 'xla_fsdp_grad_ckpt': False}, 'fsdp_transformer_layer_cls_to_wrap': None, 'accelerator_config': {'split_batches': False, 'dispatch_batches': None, 'even_batches': True, 'use_seedable_sampler': True, 'non_blocking': False, 'gradient_accumulation_kwargs': None}, 'deepspeed': {'train_batch_size': 4, 'train_micro_batch_size_per_gpu': 1, 'zero_optimization': {'stage': 3, 'offload_optimizer': {'device': 'cpu', 'pin_memory': True}, 'offload_param': {'device': 'cpu', 'pin_memory': True}, 'overlap_comm': True, 'contiguous_gradients': True, 'sub_group_size': 1000000000.0, 'reduce_bucket_size': 50000000.0, 'stage3_prefetch_bucket_size': 50000000.0, 'stage3_param_persistence_threshold': 10000.0, 'stage3_max_live_parameters': 100000000.0, 'stage3_max_reuse_distance': 100000000.0, 'stage3_gather_16bit_weights_on_model_save': True, 'round_robin_gradients': True}, 'gradient_accumulation_steps': 4, 'gradient_clipping': 1.0, 'steps_per_print': 10, 'fp16': {'enabled': False}, 'bf16': {'enabled': True}, 'activation_checkpointing': {'partition_activations': False, 'cpu_checkpointing': False, 'contiguous_memory_optimization': False, 'number_checkpoints': None, 'synchronize_checkpoint_boundary': False, 'profile': False}, 'wall_clock_breakdown': False, 'memory_breakdown': False}, 'label_smoothing_factor': 0.0, 'optim': 'adamw_torch', 'optim_args': None, 'adafactor': False, 'group_by_length': False, 'length_column_name': 'length', 'report_to': ['wandb'], 'ddp_find_unused_parameters': None, 'ddp_bucket_cap_mb': None, 'ddp_broadcast_buffers': None, 'dataloader_pin_memory': False, 'dataloader_persistent_workers': False, 'skip_memory_metrics': True, 'use_legacy_prediction_loop': False, 'push_to_hub': False, 'resume_from_checkpoint': None, 'hub_model_id': None, 'hub_strategy': 'every_save', 'hub_token': '<HUB_TOKEN>', 'hub_private_repo': None, 'hub_always_push': False, 'hub_revision': None, 'gradient_checkpointing': False, 'gradient_checkpointing_kwargs': None, 'include_inputs_for_metrics': False, 'include_for_metrics': [], 'eval_do_concat_batches': True, 'fp16_backend': 'auto', 'push_to_hub_model_id': None, 'push_to_hub_organization': None, 'push_to_hub_token': '<PUSH_TO_HUB_TOKEN>', 'mp_parameters': '', 'auto_find_batch_size': False, 'full_determinism': False, 'torchdynamo': None, 'ray_scope': 'last', 'ddp_timeout': 1800, 'torch_compile': False, 'torch_compile_backend': None, 'torch_compile_mode': None, 'include_tokens_per_second': False, 'include_num_input_tokens_seen': False, 'neftune_noise_alpha': None, 'optim_target_modules': None, 'batch_eval_metrics': False, 'eval_on_start': False, 'use_liger_kernel': False, 'liger_kernel_config': None, 'eval_use_gather_object': False, 'average_tokens_across_devices': False, 'model_init_kwargs': None, 'disable_dropout': False, 'max_prompt_length': 512, 'num_generations': 2, 'max_completion_length': 2048, 'ds3_gather_for_generation': True, 'shuffle_dataset': True, 'generation_batch_size': 2, 'steps_per_generation': 2, 'min_p': None, 'generation_kwargs': None, 'use_transformers_paged': False, 'cache_implementation': None, 'use_vllm': False, 'vllm_server_base_url': None, 'vllm_mode': 'server', 'vllm_model_impl': 'vllm', 'vllm_guided_decoding_regex': None, 'vllm_server_host': '0.0.0.0', 'vllm_server_port': 8000, 'vllm_server_timeout': 240.0, 'vllm_gpu_memory_utilization': 0.3, 'vllm_tensor_parallel_size': 1, 'beta': 0.0, 'num_iterations': 1, 'epsilon': 0.2, 'delta': None, 'epsilon_high': 0.28, 'importance_sampling_level': 'token', 'reward_weights': None, 'scale_rewards': True, 'loss_type': 'bnpo', 'mask_truncated_completions': False, 'sync_ref_model': False, 'ref_model_mixup_alpha': 0.6, 'ref_model_sync_steps': 512, 'top_entropy_quantile': 1.0, 'use_liger_loss': False, 'log_completions': False, 'num_completions_to_print': None, 'wandb_log_unique_prompts': False}
2025-08-27 17:30:15,855 INFO    MainThread:1563477 [wandb_config.py:__setitem__():154] [no run ID] config set model/num_parameters = 0 - <bound method Run._config_callback of <wandb.sdk.wandb_run.Run object at 0x7f37f7755750>>
2025-08-27 17:30:15,855 INFO    MainThread:1563477 [wandb_run.py:_config_callback():1380] config_cb model/num_parameters 0 None
2025-08-27 23:16:19,557 INFO    MainThread:1563477 [wandb_run.py:_finish():2260] finishing run gistdslab/gpt-oss-arc-hf-trl-dapo/17nljko7
2025-08-27 23:16:19,559 INFO    MainThread:1563477 [wandb_run.py:_atexit_cleanup():2459] got exitcode: 0
2025-08-27 23:16:19,560 INFO    MainThread:1563477 [wandb_run.py:_restore():2441] restore
2025-08-27 23:16:19,560 INFO    MainThread:1563477 [wandb_run.py:_restore():2447] restore done
2025-08-27 23:16:20,877 INFO    MainThread:1563477 [wandb_run.py:_restore():2441] restore
2025-08-27 23:16:20,877 INFO    MainThread:1563477 [wandb_run.py:_restore():2447] restore done
2025-08-27 23:16:20,877 ERROR   MainThread:1563477 [wandb_run.py:_atexit_cleanup():2480] Problem finishing run
Traceback (most recent call last):
  File "/home/ubuntu/gpt_oss_arc_final/main_hf_trl_dapo.py", line 612, in main_hf_trl_dapo
    trainer.train()
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/transformers/trainer.py", line 2238, in train
    return inner_training_loop(
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/transformers/trainer.py", line 2582, in _inner_training_loop
    tr_loss_step = self.training_step(model, inputs, num_items_in_batch)
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/transformers/trainer.py", line 3790, in training_step
    inputs = self._prepare_inputs(inputs)
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/trl/extras/profiling.py", line 98, in wrapper
    return func(self, *args, **kwargs)
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/trl/trainer/grpo_trainer.py", line 1279, in _prepare_inputs
    generation_batch = self._generate_and_score_completions(generation_batch)
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/trl/trainer/grpo_trainer.py", line 1577, in _generate_and_score_completions
    prompt_completion_ids = unwrapped_model.generate(
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/peft/peft_model.py", line 1973, in generate
    outputs = self.base_model.generate(*args, **kwargs)
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 116, in decorate_context
    return func(*args, **kwargs)
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/transformers/generation/utils.py", line 2617, in generate
    result = self._sample(
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/transformers/generation/utils.py", line 3601, in _sample
    outputs = model_forward(**model_inputs, return_dict=True)
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/transformers/utils/generic.py", line 959, in wrapper
    output = func(self, *args, **kwargs)
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/transformers/models/gpt_oss/modeling_gpt_oss.py", line 658, in forward
    outputs: MoeModelOutputWithPast = self.model(
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/transformers/utils/generic.py", line 1083, in wrapper
    outputs = func(self, *args, **kwargs)
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/transformers/models/gpt_oss/modeling_gpt_oss.py", line 491, in forward
    hidden_states = decoder_layer(
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/transformers/modeling_layers.py", line 93, in __call__
    return super().__call__(*args, **kwargs)
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/transformers/models/gpt_oss/modeling_gpt_oss.py", line 370, in forward
    hidden_states, _ = self.mlp(hidden_states)  # diff with llama: router scores
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/transformers/models/gpt_oss/modeling_gpt_oss.py", line 158, in forward
    routed_out = self.experts(hidden_states, router_indices=router_indices, routing_weights=router_scores)
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/transformers/models/gpt_oss/modeling_gpt_oss.py", line 111, in forward
    out = gated_output @ self.down_proj[expert_idx] + self.down_proj_bias[expert_idx]
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1918, in __getattr__
    def __getattr__(self, name: str) -> Any:
KeyboardInterrupt

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/wandb/sdk/wandb_run.py", line 2471, in _atexit_cleanup
    self._on_finish()
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/wandb/sdk/wandb_run.py", line 2723, in _on_finish
    wait_with_progress(
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/wandb/sdk/mailbox/wait_with_progress.py", line 24, in wait_with_progress
    return wait_all_with_progress(
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/wandb/sdk/mailbox/wait_with_progress.py", line 87, in wait_all_with_progress
    return asyncio_compat.run(progress_loop_with_timeout)
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/wandb/sdk/lib/asyncio_compat.py", line 30, in run
    return future.result()
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/concurrent/futures/_base.py", line 458, in result
    return self.__get_result()
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/concurrent/futures/_base.py", line 403, in __get_result
    raise self._exception
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/wandb/sdk/lib/asyncio_compat.py", line 74, in run
    return asyncio.run(self._run_or_cancel(fn))
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/wandb/sdk/lib/asyncio_compat.py", line 98, in _run_or_cancel
    return fn_task.result()
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/wandb/sdk/mailbox/wait_with_progress.py", line 75, in progress_loop_with_timeout
    with asyncio_compat.cancel_on_exit(display_progress()):
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/contextlib.py", line 142, in __exit__
    next(self.gen)
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/wandb/sdk/lib/asyncio_compat.py", line 211, in cancel_on_exit
    raise exception
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/wandb/sdk/wandb_run.py", line 2699, in _display_finish_stats
    async with asyncio_compat.open_task_group() as task_group:
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/contextlib.py", line 206, in __aexit__
    await anext(self.gen)
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/wandb/sdk/lib/asyncio_compat.py", line 193, in open_task_group
    await task_group._wait_all()
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/wandb/sdk/lib/asyncio_compat.py", line 162, in _wait_all
    raise exc
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/wandb/sdk/wandb_run.py", line 2679, in loop_update_printer
    progress_printer.update(
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/wandb/sdk/lib/progress.py", line 108, in update
    self._update_operation_stats(
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/wandb/sdk/lib/progress.py", line 143, in _update_operation_stats
    self._printer.display(line)
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/wandb/sdk/lib/printer.py", line 318, in display
    self._display_fn_mapping(level)(text)
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/wandb/errors/term.py", line 172, in termlog
    _log(
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/wandb/errors/term.py", line 358, in _log
    click.echo(string, file=sys.stderr, nl=newline)
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/click/utils.py", line 321, in echo
    file.write(out)  # type: ignore
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "/data/miniforge3/envs/gpt_oss_rl/lib/python3.10/site-packages/wandb/sdk/lib/console_capture.py", line 168, in write_with_callbacks
    n = orig_write(s)
BrokenPipeError: [Errno 32] Broken pipe
2025-08-27 23:16:21,877 INFO    MsgRouterThr:1563477 [mailbox.py:close():129] [no run ID] Closing mailbox, abandoning 1 handles.
